esp/services/ws_workunits/ws_workunitsService.cpp

```diff
@@ -2847,6 +2847,17 @@ bool CWsWorkunitsEx::onWUResult(IEspContext &context, IEspWUResultRequest &req,
         const char* logicalName = req.getLogicalName();
         const char* clusterName = req.getCluster();
         const char* resultName = req.getResultName();
+        bool cacheResult = false;
+        {
+            Owned<IWorkUnitFactory> wf = getWorkUnitFactory(context.querySecManager(), context.queryUser());
+            Owned<IConstWorkUnit> cw = wf->openWorkUnit(wuid, false);
```

Comment:
When getWsWuResult is called maybe it could provide the state?  So you don't have to open the workunit again?